engine/src/player.ts

```diff
@@ -595,11 +596,11 @@ export default class Player extends EventEmitter {
     }
   }
 
-  receivePassIncome() {
+  receivePassIncome(consume: (rewards: Reward[], source: EventSource) => void) {
     // this is for pass tile income (e.g. round boosters, adv tiles)
     for (const event of this.events[Operator.Pass]) {
       const times = this.eventConditionCount(event.condition);
-      this.gainRewards(
+      consume(
```

Comment:
Why this change? Is `receivePassIncome` called somewhere else?

Reply:
yes, in [vp estimation](https://github.com/boardgamers/gaia-project/pull/18/files#diff-d81e83bc496b6ae3460f7a869ee73a1e9011102be8df59ef70f5d81b671cd028R84)

Reply:
I understand, but I'd like to avoid making the engine code more complicated for things specific to the viewer. Here it's a bit hard to understand the reasoning without at least a comment, and the style changes from the rest of the engine code. 

Luckily, there's an alternative which does not add too much complexity. When simulating VP, instead of cloning the player data, you can clone the player:

```
const json = JSON.parse(JSON.stringify(player));
delete json.federationCache; // otherwise we need to pass the map when loading
const clone = Player.fromData(json, null, 0);
```

That way you can just do `clone.receivePassIncome()` (and `clone.data.finalHandling()`, ...)

Reply:
tried it - but got a strange error - it's in the last commit. can you help?

Reply:
fixed it - I think it was due to the lazy loading of chart.